programs/launcher/main.cpp

```diff
@@ -747,22 +905,35 @@ launcher_def::kill (launch_modes mode, string sig_opt) {
 
 void
 launcher_def::start_all (string &gts, launch_modes mode) {
-  if (mode == LM_NONE)
+  switch (mode) {
+  case LM_NONE:
     return;
-
-  if (random_start) {
-    // recompute start order - implemenation deferred
-  } else {
-    for (auto &node : network.nodes) {
-      if (mode != LM_ALL) {
-        if ((mode == LM_LOCAL && node.second.remote) ||
-            (mode == LM_REMOTE && !node.second.remote)) {
-          continue;
+  case LM_VERIFY:
+    //validate configuration, report findings, exit
+    return;
+  case LM_NAMED : {
+    try {
+      auto node = network.nodes.find(launch_name);
+      launch(*node->second.instance, gts);
+    } catch (...) {
+      // failed to start.
+    }
+    break;
+  }
+  case LM_ALL:
+  case LM_REMOTE:
+  case LM_LOCAL: {
+    for (auto &h : host_map.bindings ) {
+      if (mode == LM_ALL ||
+          (h.second.is_local() ? mode == LM_LOCAL : mode == LM_REMOTE)) {
+        for (auto &inst : h.second.instances) {
+          launch (inst, gts);
```

Comment:
`launch()` above is wrapped in try{}catch{}.  Shouldn't this invocation also be wrapped?

Reply:
I can add that.